pkg/rtc/rtc_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package rtc
+
+import (
+	"os"
+	"runtime"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/testutil"
+	"golang.org/x/sys/unix"
+)
+
+type testSyscalls struct{}
+
+func (tsc testSyscalls) ioctlGetRTCTime(fd int) (*unix.RTCTime, error) {
+	return &unix.RTCTime{}, nil
+}
+
+func (tsc testSyscalls) ioctlSetRTCTime(fd int, time *unix.RTCTime) error {
+	return nil
+}
+
+func OpenMockRTC() (*RTC, error) {
+	var tsc testSyscalls
+	f, err := os.CreateTemp("", "rtc-*")
+	if err != nil {
+		return nil, err
+	}
+	return &RTC{
+		f,
+		tsc,
+	}, nil
+}
+
+func TestOpenRTC(t *testing.T) {
+	for _, tt := range []struct {
+		name string
+	}{
+		{
+			name: "open mock rtc",
+		},
+		{
+			name: "open real rtc",
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.name == "open mock rtc" {
+				rtc, err := OpenMockRTC()
+				if err != nil {
+					t.Errorf("Expected: nil Got: %v", err)
+				}
+				rtc.Close()
+				if err := os.Remove(rtc.file.Name()); err != nil {
+					t.Errorf("Failed removing tempfile: %v", err)
+				}
+			}
+			if tt.name == "open real rtc" {
+				testutil.SkipIfNotRoot(t)
+				// The u-root amd64 VM does not seem to have a RTC device
```

Comment:
Are you sure or are we just missing a kernel CONFIG option?

Reply:
Probably both is the case, I tried adding a Qemu option to add an RTC device but the test still failed there stating no RTC was found in the system